src/functions/github/handler.ts

```diff
@@ -0,0 +1,206 @@
+import { Buffer } from "node:buffer";
+import { URL } from "node:url";
+import { logger } from "@yuudachi/framework";
+import {
+	type AnyThreadChannel,
+	type Message,
+	type VoiceChannel,
+	PermissionFlagsBits,
+	ThreadAutoArchiveDuration,
+	codeBlock,
+	ChannelType,
+} from "discord.js";
+import { request } from "undici";
+import { arrayEllipsis, stringArrayLength } from "../../util/array.js";
+import { URL_REGEX } from "../../util/constants.js";
+import { GistGitHubUrlRegex, GitHubUrlLinesRegex, NormalGitHubUrlRegex } from "./regex.js";
+import { formatLine, generateHeader, resolveFileLanguage, resolveLines, validateFileSize } from "./utils.js";
+
+const SAFE_BOUNDARY = 50;
+
+enum GitHubUrlType {
+	Normal,
+	Gist,
+	Diff,
+}
+
+const validators = [
+	{
+		type: GitHubUrlType.Normal,
+		regex: NormalGitHubUrlRegex,
+		converter: (url: string): string =>
+			url
+				.replace(">", "")
+				.replace("github.com", "raw.githubusercontent.com")
+				.replace(/\/(?:blob|(?:blame))/, ""),
+	},
+	{
+		type: GitHubUrlType.Gist,
+		regex: GistGitHubUrlRegex,
+		converter: (url: string): string | null => {
+			// eslint-disable-next-line unicorn/no-unsafe-regex
+			const { user, id, opts } = new RegExp(GistGitHubUrlRegex, "").exec(url.replace(/(?:-L\d+)+/, ""))!.groups!;
+
+			if (!user || !id || !opts) {
+				return null;
+			}
+
+			const fileName = opts.split("-").slice(0, -1).join("-");
+			const extension = opts.split("-").pop();
+
+			return `https://gist.githubusercontent.com/${user}/${id}/raw/${fileName}.${extension}`;
+		},
+	},
+];
+
+export async function handleGithubUrls(message: Message<true>) {
+	const urls = new Set(message.content.matchAll(URL_REGEX));
+	if (!urls.size) return;
+
+	const matches = Array.from(urls).map(([url]) => {
+		const match = validators.find((validator) => validator.regex.exec(url!));
+		if (!match) return null;
+
+		return {
+			regexMatch: new RegExp(match.regex, "").exec(url!),
```

Comment:
Return null if the regex match doesn't match
```suggestion
		if (!match) return null;

		const regexMatch = new RegExp(match.regex, "").exec(url!);
		if (!regexMatch) return null;

		return {
			regexMatch,
```